src/Microsoft.Restier.Core/Conventions/ConventionBasedEntitySetFilter.cs

```diff
@@ -22,23 +22,33 @@ private ConventionBasedEntitySetFilter(Type targetType)
             this.targetType = targetType;
         }
 
+        public IQueryExpressionFilter Inner { get; set; }
+
         /// <inheritdoc/>
         public static void ApplyTo(
             IServiceCollection services,
             Type targetType)
         {
             Ensure.NotNull(services, "services");
             Ensure.NotNull(targetType, "targetType");
-            services.CutoffPrevious<IQueryExpressionFilter>(new ConventionBasedEntitySetFilter(targetType));
+            services.ChainPrevious<IQueryExpressionFilter>(next => new ConventionBasedEntitySetFilter(targetType)
+            {
+                Inner = next,
+            });
         }
 
         /// <inheritdoc/>
         public Expression Filter(QueryExpressionContext context)
         {
             Ensure.NotNull(context, "context");
-            if (context.ModelReference == null)
+
+            if (Inner != null)
             {
-                return null;
+                var innerFilteredExpression = Inner.Filter(context);
+                if (innerFilteredExpression != null)
```

Comment:
Now if inner filter has any logic, it will filter and return a not null expression, or it will return null just like what current filter does.
The logic here means if inner has some logic do the filter, then just return that one, or it will continue the processing with existing filter.

DO you want me to change inner filter logic make it will return original expression even it does nothing?